### PR TITLE
Add macos package builder

### PR DESCRIPTION
**Pull request by VijayanB**
Added script which will generate macos pkg from opensearch-cli
binary. To generate pkg, copy opensearch-cli binary into
application folder. Execute script "build-macos.sh" by following
usuage instructions.
pkg will be generated inside target/pkg directory.

Signed-off-by: Vijayan Balasubramanian <balasvij@amazon.com>

### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has documentation added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/opensearch-cli/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by vamshin — DISMISSED** [review]
LGTM! Thanks

File: macos-pkg-builder/build-macos.sh [hunk 1]
```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
```
Comment (jmazanec15) [inline macos-pkg-builder/build-macos.sh:1]: Do we need license at top of here?

File: macos-pkg-builder/build-macos.sh [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+copyDarwinDirectory(){
+  createInstallationDirectory
+  cp -r $SCRIPTPATH/darwin ${TARGET_DIRECTORY}/
+  chmod -R 755 ${TARGET_DIRECTORY}/darwin/scripts
+  chmod -R 755 ${TARGET_DIRECTORY}/darwin/Resources
+  chmod 755 ${TARGET_DIRECTORY}/darwin/Distribution
+}
+
+copyBuildDirectory() {
```
Comment (jmazanec15) [inline macos-pkg-builder/build-macos.sh:101]: Why are some functions prefixed with "function" and others are not?

File: macos-pkg-builder/build-macos.sh [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    echo "${LOG_PREFIX}[INFO]" $1
+}
+
+log_error() {
+    echo "${LOG_PREFIX}[ERROR]" $1
+}
+
+deleteInstallationDirectory() {
+    log_info "Cleaning $TARGET_DIRECTORY directory."
+    rm -rf $TARGET_DIRECTORY
```
Comment (jmazanec15) [inline macos-pkg-builder/build-macos.sh:73]: Why not make target dir an argument, as is done in previous functions?

File: macos-pkg-builder/build-macos.sh [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  cp -r $SCRIPTPATH/darwin ${TARGET_DIRECTORY}/
+  chmod -R 755 ${TARGET_DIRECTORY}/darwin/scripts
+  chmod -R 755 ${TARGET_DIRECTORY}/darwin/Resources
+  chmod 755 ${TARGET_DIRECTORY}/darwin/Distribution
+}
+
+copyBuildDirectory() {
+    sed -i '' -e 's/__VERSION__/'${VERSION}'/g' ${TARGET_DIRECTORY}/darwin/scripts/postinstall
+    sed -i '' -e 's/__PRODUCT__/'${PRODUCT}'/g' ${TARGET_DIRECTORY}/darwin/scripts/postinstall
+    chmod -R 755 ${TARGET_DIRECTORY}/darwin/scripts/postinstall
```
Comment (jmazanec15) [inline macos-pkg-builder/build-macos.sh:104]: Dont we already do this in copyDarwinDirectory?

File: macos-pkg-builder/build-macos.sh (hunk 1 above)
Comment (VijayanB) [reply macos-pkg-builder/build-macos.sh:1]: Ack

File: macos-pkg-builder/build-macos.sh (hunk 2 above)
Comment (VijayanB) [reply macos-pkg-builder/build-macos.sh:101]: removed all function prefix for consistency

File: macos-pkg-builder/build-macos.sh (hunk 4 above)
Comment (VijayanB) [reply macos-pkg-builder/build-macos.sh]: Ack. Removed it here.

File: macos-pkg-builder/build-macos.sh (hunk 3 above)
Comment (VijayanB) [reply macos-pkg-builder/build-macos.sh:85]: Ack. Added as argument.

**Review by jmazanec15 — APPROVED** [review]

**Review by naveentatikonda — APPROVED** [review]

**Pull request merged**